src/libraries/System.Web.HttpUtility/src/System/Web/Util/HttpEncoder.cs

```diff
@@ -240,14 +246,14 @@ internal static string JavaScriptStringEncode(string? value)
                 decodedBytes[decodedBytesCount++] = b;
             }
 
-            if (decodedBytesCount < decodedBytes.Length)
+            if (decodedBytesCount < count)
```

Comment:
Both of these blocks allocate a new array for the subsequence.
You can delete this `if` block and let the `ToArray` below handle it.